src/hotspot/share/oops/fieldInfo.hpp

```diff
@@ -25,11 +25,29 @@
 #ifndef SHARE_OOPS_FIELDINFO_HPP
 #define SHARE_OOPS_FIELDINFO_HPP
 
+#include "memory/allocation.hpp"
+#include "memory/metadataFactory.hpp"
 #include "oops/constantPool.hpp"
 #include "oops/symbol.hpp"
```

Comment:
Since you added an inline.hpp function can you move the functions that rely on including constantPool.hpp, symbol.hpp and metadataFactory.hpp into the inline.hpp file?